create structure for internal docs (#276)
packit/packit.dev: content/internal-docs/patch-metadata.md

[dist2src](https://github.com/packit/dist-git-to-source-git) and `packit init`.

Ideally maintainers would just commit changes to source-git repos and have
`%autosetup` to apply all the patches during `%prep`.

Example of patch metadata in one of commits in systemd source-git for CentOS
Stream 8:
```
$ git log HEAD

commit 38e6b5b3059410530e0d5287de595cbf4574988b (HEAD -> c8s, upstream/c8s)
Author: Lennart Poettering <lennart@poettering.net>
Date:   Mon Feb 4 10:23:43 2019 +0100

    pam-systemd: use secure_getenv() rather than getenv()
    
    And explain why in a comment.
    
    (cherry picked from commit 83d4ab55336ff8a0643c6aa627b31e351a24040a)
    
    CVE-2019-3842
    
    Resolves: #1687514
    
    patch_name: 0563-pam-systemd-use-secure_getenv-rather-than-getenv.patch
    present_in_specfile: true
    location_in_specfile: 563
    squash_commits: true

diff --git a/src/login/pam_systemd.c b/src/login/pam_systemd.c
...
```

You can see the patch metadata are stored in the commit message on the last 4
lines.

## Metadata

The metadata are stored in commit messages and have a key-value format parsed as
yaml. The list of keys follows.

### `patch_name`
**Type**: str  

> **Default**: the default comes from `git-format-patch`: `By default, each
> output file is numbered sequentially from 1, and uses the first line of the
> commit message`  

Reviewer: ```suggestion
**Default**: the default comes from `git-format-patch`: "By default, each
output file is numbered sequentially from 1, and uses the first line of the
commit message"   
```

**Content**: file name of the patch  
**Example**: "my-fancy.patch"  

> Patch file generated from the specific commit should bear this file name. This

Reviewer 2: I'm a bit struggling with the first sentence. Maybe this would make it a bit easier for me:
```suggestion
Patch file generated from the commit will have this name. This
```

these - our tooling does that:
[dist2src](https://github.com/packit/dist-git-to-source-git) and `packit init`.

Ideally maintainers would just commit changes to source-git repos and have
`%autosetup` to apply all the commits.

## Metadata

The metadata is stored in commit messages and has a key-value format parsed as
yaml. The list of keys follows.

### `patch_name`
**Type**: str  
**Content**: file name of the patch  
**Example**: "my-fancy.patch"  
Patch file generated from the specific commit should bear this file name. This
is useful when a patch is already defined in the spec file and we need to make
the patch file match that `Patch` spec file entry.

### `location_in_specfile`
**Type**: int  
**Example**: 2  
Represents the ID of the patch entry in the spec file. The field **is not being
used** right now when populating spec file with patch entries - it's only
informational and does not change the patch process in any way.

### `description`
**Type**: str  
**Example**: "This patch is cherry-picked from upstream commit ea45faaa and
resolves build failures on arm."  
Human-friendly description of the patch file to be put above the spec file entry.

### `present_in_specfile`

> **Type**: bool  

Reviewer: What are the default values for these metadata fields?
